### PR TITLE
PAR-520

### PR DESCRIPTION
**Pull request by aYoussef**

### CODE REVIEW

**Comment by oaosman84** [comment]
LGTM

**Comment by oaosman84** [comment]
@kfateem if you want to have a look before we merge, this is the solution for push notifications when the app isn't open on iOS

**Comment by kfateem** [comment]
Please see comment on:
https://github.com/Parlio/web/pull/349

**Comment by kfateem** [comment]
Thanks for moving to an event-driven paradigm.

Please register and unregister for "WEBCLIENTREADY" in the register/unregister methods - not notificationReceived.

When WEBCLIENTREADY check if you have a notification then handle.

**Comment by kfateem** [comment]
@aYoussef after our discussion please:
1) add an observer "WEBCLIENTREADY" in AppDelegate+notification.m swizzled_init with callback to onWebClientReady()
2) Maintain a boolean flag isWebClientReady
3) in onWebClientReady()
    i) updated isWebClientReady to true
   ii) call notificationReceived
  iii) unregister the observer
4) modify notificationReceived to check isWebClientReady flag before attempting to communicate with web:

```
 if (notificationMessage && self.callback && isWebClientReady) {
...
}
```

File: www/PushNotification.js [hunk 1]
```diff
@@ -64,6 +64,10 @@ PushNotification.prototype.setApplicationIconBadgeNumber = function(successCallb
     cordova.exec(successCallback, errorCallback, "PushPlugin", "setApplicationIconBadgeNumber", [{badge: badge}]);
 };
 
+PushNotification.prototype.handlePendingNotifications = function(successCallback, errorCallback) {
```
Comment (kfateem) [inline www/PushNotification.js]: handlePendingNotifications used anywhere?

**Comment by kfateem** [comment]
1 comment otherwise LGTM :+1: 
Thank you for handling!

**Pull request merged**